### PR TITLE
cleanup `_getPath`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/property_get.js [hunk 1]
```diff
@@ -74,24 +68,20 @@ export function _getPath(root, path) {
   let parts = path.split('.');
 
   for (let i = 0; i < parts.length; i++) {
-    if (!isGettable(obj)) {
+    if (obj === undefined || obj === null || obj.isDestroyed) {
```
Comment (rwjblue) [inline packages/ember-metal/lib/property_get.js]: I don't think we want to hard code `.isDestroyed` check here. In theory it would be better to `let m = peekMeta(obj); m.isDestroyed()`.

File: packages/ember-metal/lib/property_get.js [hunk 2]
```diff
@@ -5,12 +5,6 @@
 import { assert } from 'ember-debug';
 import { isPath } from './path_cache';
 
-const ALLOWABLE_TYPES = {
```
Comment (rwjblue) [inline packages/ember-metal/lib/property_get.js]: Why is this removed?

Comment (bekzod) [reply packages/ember-metal/lib/property_get.js]: why to have it :P, even if other types passed `get` will return `undefined`

File: packages/ember-metal/lib/property_get.js (hunk 1 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_get.js]: it is how done before https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/property_get.js#L83 <- just moved this check to top
I think checking `isDestroyed` with `meta` worth another PR since hardcoded `isDestroyed` check is done in more places

File: packages/ember-metal/lib/property_get.js (hunk 2 above)
Comment (rwjblue) [reply packages/ember-metal/lib/property_get.js]: Generally speaking, its best to review the git history to understand a given bit of code's purpose.

* https://github.com/emberjs/ember.js/issues/13444
* https://github.com/emberjs/ember.js/pull/13449
* https://github.com/emberjs/ember.js/pull/13461

Specifically answering your question though is https://github.com/emberjs/ember.js/pull/13449#r62256688.

Comment (bekzod) [reply packages/ember-metal/lib/property_get.js]: added test for those cases if something missing will add it, I think all cases handled correctly

File: packages/ember-metal/lib/property_get.js (hunk 1 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_get.js]: I will do follow up PR to use `meta.isDestroyed()` does it sound good ? or should I do it with this PR

Comment (rwjblue) [reply packages/ember-metal/lib/property_get.js]: Ya, sounds good to do separately

**Comment by bekzod** [comment]
ping, any objections to this ?

File: packages/ember-metal/lib/property_get.ts [hunk 3]
```diff
@@ -178,24 +172,16 @@ export function _getPath<T extends object>(root: T, path: string): any {
   let parts = path.split('.');
 
   for (let i = 0; i < parts.length; i++) {
-    if (!isGettable(obj)) {
+    if (obj === undefined || obj === null || (obj as MaybeHasIsDestroyed).isDestroyed) {
```
Comment (bekzod) [inline packages/ember-metal/lib/property_get.ts]: maybe move `(obj as MaybeHasIsDestroyed).isDestroyed ` to `get` itself
